### ISSUE TITLE
[BUG] Seelen UI is not responding on two monitors or more

### ISSUE THREAD

**Issue by CoolCoderSJ** [eythaann/Seelen-UI#19]
**Describe the bug**
The UI just won't start up anymore. When opening the app it's completely non-responsive, and none of the UI features (menu bar, dock, etc.) load.

I've tried the exe from the releases as well as microsoft store and uninstalled + reinstalled multiple times.

**To Reproduce**
1. Install Seelen UI
2. app hangs

**Expected behavior**
Should not hang.

**Screenshots**
![image](https://github.com/user-attachments/assets/bb527b92-442f-41ad-b069-7d517c8506ed)

**Comment by eythaann** [comment]
on discord we are talking about this issue for some user the app crash, for other ones it works without any problems.

**Comment by CoolCoderSJ** [comment]
yes- 1.9.4, and I have tried running as admin as well
oh I can join the discord as well then

**Comment by mutebuddha** [comment]
good day. for me - its just launch icon and closes.

**Comment by eythaann** [comment]
Replicate: The app frozen when try to start it and have two or more monitors.

**Comment by eythaann** [comment]
> good day. for me - its just launch icon and closes.

@mutebuddha do you have more than one monitor to be sure is the same issue?

**Comment by Tsubajashi** [comment]
@eythaann same issue as @mutebuddha , 2 monitors. first monitor is 2560x1440@144hz@100% and second one is 3840x2160@60hz@200%.

**Comment by eythaann** [comment]
> @eythaann same issue as @mutebuddha , 2 monitors. first monitor is 2560x1440@144hz@100% and second one is 3840x2160@60hz@200%.

can you try disconnect one monitor and retry it? be sure the no process called Seelen is open on Task Manager before try it.

**Comment by Tsubajashi** [comment]
@eythaann with one monitor only plugged in, after a fresh restart, it still doesnt work. same behaviour as previously. Stock Windows 11 24H2, on a R9 7950x and 2 4090s, although just one of them is used for monitors.

**Comment by mutebuddha** [comment]
> > good day. for me - its just launch icon and closes.
> 
> @mutebuddha do you have more than one monitor to be sure is the same issue?

nope, one monitor. the logo loads, the logo hides, that's it. When first launched, the program applied "automatic hiding of the taskbar".

**Comment by eythaann** [comment]
This bug should be fixed from: e88e9223798801a3d2254d2a3b0de02fc36539de
And be deployed on v1.9.5 release.

**Comment by strepito66** [comment]
Hello, I have the same issues and for me it's still there in 1.9.5

**Comment by mutebuddha** [comment]
> This bug should be fixed from: [e88e922](https://github.com/eythaann/Seelen-UI/commit/e88e9223798801a3d2254d2a3b0de02fc36539de) And be deployed on v1.9.5 release.

https://github.com/user-attachments/assets/6d12c6ed-ff7b-4862-b2cf-455f49735098

**Comment by Tsubajashi** [comment]
@eythaann the same issue occurs on my end. are there any logs that i can send to you to better determine the issue at hand?

**Comment by eythaann** [comment]
I see it we have two bugs, the one that froze the app was fixed, but there is other wants that crash and close the app.

**Comment by eythaann** [comment]
> @eythaann the same issue occurs on my end. are there any logs that i can send to you to better determine the issue at hand?

sure the log file is on: `C:\Users\{your user}\AppData\Local\com.seelen.seelen-ui\logs`

**Comment by Tsubajashi** [comment]
> > @eythaann the same issue occurs on my end. are there any logs that i can send to you to better determine the issue at hand?
> 
> sure the log file is on: `C:\Users\{your user}\AppData\Local\com.seelen.seelen-ui\logs`

[2024-08-17][08:46:57][seelen_ui][INFO] ───────────────────── Starting Seelen ─────────────────────
[2024-08-17][08:46:57][seelen_ui::modules::cli][INFO] TCP server listening on 127.0.0.1:53228
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Ensuring folders
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Initializing Seelen
[2024-08-17][08:46:57][seelen_ui::state::application][TRACE] Seelen UI Data Watcher started!
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Showing update notification window
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Show settings window
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Show settings window
[2024-08-17][08:46:57][seelen_ui::seelen][TRACE] Enumerating Monitors
[2024-08-17][08:46:57][seelen_ui::seelen_bar][INFO] Creating fancy-toolbar/DISPLAY1
[2024-08-17][08:46:58][reqwest::connect][DEBUG] starting new connection: https://github.com/
[2024-08-17][08:46:58][seelen_ui::seelen_weg][INFO] Creating seelenweg/DISPLAY1
[2024-08-17][08:46:58][reqwest::async_impl::client][DEBUG] redirecting 'https://github.com/eythaann/Seelen-UI/releases/latest/download/latest.json' to 'https://github.com/eythaann/Seelen-UI/releases/download/v1.9.5/latest.json'
[2024-08-17][08:46:58][reqwest::async_impl::client][DEBUG] redirecting 'https://github.com/eythaann/Seelen-UI/releases/download/v1.9.5/latest.json' to 'https://objects.githubusercontent.com/github-production-release-asset-2e65be/758623687/cde53d70-cffe-41ff-8346-ef1441464d72?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240817%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240817T084651Z&X-Amz-Expires=300&X-Amz-Signature=3b54a2549962e9f42c4ed053e64deb2e8045ba794d9285425c280a1d15d0f8db&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=758623687&response-content-disposition=attachment%3B%20filename%3Dlatest.json&response-content-type=application%2Foctet-stream'
[2024-08-17][08:46:58][reqwest::connect][DEBUG] starting new connection: https://objects.githubusercontent.com/
[2024-08-17][08:46:59][seelen_ui::seelen_bar][INFO] Creating fancy-toolbar/DISPLAY2
[2024-08-17][08:46:59][seelen_ui::seelen_weg][INFO] Creating seelenweg/DISPLAY2
[2024-08-17][08:47:00][seelen_ui::tray][TRACE] registering tray icon
[2024-08-17][08:47:00][seelen_ui::seelen][TRACE] Killing AHK shortcuts
[2024-08-17][08:47:00][seelen_ui::seelen][TRACE] Starting AHK shortcuts
[2024-08-17][08:47:00][seelen_ui::seelen][TRACE] AHK shortcuts started successfully

not sure how much these logs bring to the table, but there you go. :) tested with 1.9.5 on 2 monitors

**Comment by eythaann** [comment]
someone can confirm if this issue is solved on the last version? (v1.9.6)

**Comment by JotaRata** [comment]
it's still hanging in the latest version

**Comment by mutebuddha** [comment]
yep. its launch! :)
but where is tray icons?

**Comment by mutebuddha** [comment]
> someone can confirm if this issue was solved on the last version? (v1.9.6)

yep. its launch! :)
but where is tray icons?

**Comment by Tsubajashi** [comment]
it does run now, but each time i want to switch a setting, it hangs and crashes. doesnt even seem to matter which setting i change.

```
[2024-08-18][10:06:46][seelen_ui][INFO] ───────────────────── Starting Seelen ─────────────────────
[2024-08-18][10:06:46][seelen_ui::modules::cli][INFO] TCP server listening on 127.0.0.1:57322
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Ensuring folders
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Initializing Seelen
[2024-08-18][10:06:46][seelen_ui::state::application][TRACE] Seelen UI Data Watcher started!
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Showing update notification window
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Show settings window
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Show settings window
[2024-08-18][10:06:46][seelen_ui::seelen][TRACE] Enumerating Monitors
[2024-08-18][10:06:46][seelen_ui::seelen_bar][INFO] Creating fancy-toolbar/DISPLAY1
[2024-08-18][10:06:47][reqwest::connect][DEBUG] starting new connection: https://github.com/
[2024-08-18][10:06:47][seelen_ui::seelen_weg][INFO] Creating seelenweg/DISPLAY1
[2024-08-18][10:06:47][reqwest::async_impl::client][DEBUG] redirecting 'https://github.com/eythaann/Seelen-UI/releases/latest/download/latest.json' to 'https://github.com/eythaann/Seelen-UI/releases/download/v1.9.6/latest.json'
[2024-08-18][10:06:47][reqwest::async_impl::client][DEBUG] redirecting 'https://github.com/eythaann/Seelen-UI/releases/download/v1.9.6/latest.json' to 'https://objects.githubusercontent.com/github-production-release-asset-2e65be/758623687/02c5dded-20b5-48b2-b13f-e0ec25b702ab?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240818%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240818T100446Z&X-Amz-Expires=300&X-Amz-Signature=191d44d5911b84b6e40b9cb76efc59d0fd80a78742e8d8fd6d2bc98f12e67647&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=758623687&response-content-disposition=attachment%3B%20filename%3Dlatest.json&response-content-type=application%2Foctet-stream'
[2024-08-18][10:06:47][reqwest::connect][DEBUG] starting new connection: https://objects.githubusercontent.com/
[2024-08-18][10:06:48][seelen_ui::seelen_bar][INFO] Creating fancy-toolbar/DISPLAY2
[2024-08-18][10:06:49][seelen_ui::seelen_weg][INFO] Creating seelenweg/DISPLAY2
[2024-08-18][10:06:49][seelen_ui::tray][TRACE] registering tray icon
[2024-08-18][10:06:49][seelen_ui::seelen][TRACE] Killing AHK shortcuts
[2024-08-18][10:06:49][seelen_ui::seelen][TRACE] Starting AHK shortcuts
[2024-08-18][10:06:49][seelen_ui::seelen][TRACE] AHK shortcuts started successfully
[2024-08-18][10:06:50][seelen_ui::modules::system_settings::infrastructure][TRACE] Registering colors events
[2024-08-18][10:06:50][seelen_ui::modules::tray::infrastructure][TRACE] Registering tray events
[2024-08-18][10:06:50][seelen_ui::modules::network::infrastructure][TRACE] Registering network events
[2024-08-18][10:06:50][seelen_ui::modules::power::infrastructure][TRACE] Registering system power events
[2024-08-18][10:06:50][seelen_ui::modules::notifications::application][DEBUG] Failed to use NotificationChanged: Error { code: HRESULT(0x80070490), message: "Element nicht gefunden." }, spawning thread instead
[2024-08-18][10:06:50][seelen_ui::modules::network::infrastructure][TRACE] Stop scanning networks
[2024-08-18][10:06:50][seelen_ui::modules::media::infrastructure][TRACE] Registering media events
[2024-08-18][10:06:50][seelen_ui::modules::notifications::infrastructure][TRACE] Registering notifications events
[2024-08-18][10:06:50][seelen_ui::seelen_weg::icon_extractor][TRACE] Extracting icon for "ONENOTEM.EXE"
[2024-08-18][10:06:52][seelen_ui::modules::network::infrastructure][TRACE] Stop scanning networks
[2024-08-18][10:06:52][seelen_ui::seelen_weg::icon_extractor][TRACE] Extracting icon for "ONENOTEM.EXE"
[2024-08-18][10:06:53][seelen_ui::seelen][DEBUG] Seelen UI ready in: 7.46s
[2024-08-18][10:06:53][seelen_ui::seelen][TRACE] Enumerating windows
[2024-08-18][10:06:53][seelen_ui::seelen_weg][TRACE] Adding 22678556 <=> "Settings"
[2024-08-18][10:06:53][seelen_ui::seelen_weg][TRACE] Adding 5374466 <=> "Release Seelen UI v1.9.6 · eythaann/Seelen-UI und 7 weitere Seiten - Persönlich – Microsoft\u{200b} Edge"
[2024-08-18][10:06:53][seelen_ui::seelen_weg][TRACE] Adding 3277588 <=> "Downloads – Datei-Explorer"
[2024-08-18][10:06:53][seelen_ui::seelen_weg][TRACE] Adding 2032134 <=> "Einstellungen"
[2024-08-18][10:06:53][seelen_ui::seelen_weg][TRACE] Adding 132522 <=> "Steam"
[2024-08-18][10:06:53][seelen_ui::hook][TRACE] Registering Windows and Virtual Desktop Hooks
[2024-08-18][10:07:13][seelen_ui::state::application][INFO] Seelen Settings changed: ["C:\\Users\\Tsubajashi\\AppData\\Roaming\\com.seelen.seelen-ui\\settings.json"]
[2024-08-18][10:07:13][seelen_ui::seelen][TRACE] Killing AHK shortcuts
[2024-08-18][10:07:13][seelen_ui::seelen][TRACE] Starting AHK shortcuts
[2024-08-18][10:07:13][seelen_ui::seelen][TRACE] AHK shortcuts started successfully
[2024-08-18][10:07:13][seelen_ui::state::application][INFO] Seelen Settings changed: ["C:\\Users\\Tsubajashi\\AppData\\Roaming\\com.seelen.seelen-ui\\settings.json"]
[2024-08-18][10:07:13][seelen_ui::state::application][INFO] Specific App Configuration changed: ["C:\\Users\\Tsubajashi\\AppData\\Roaming\\com.seelen.seelen-ui\\applications.yml"]
[2024-08-18][10:07:13][seelen_ui::state::application][INFO] Specific App Configuration changed: ["C:\\Users\\Tsubajashi\\AppData\\Roaming\\com.seelen.seelen-ui\\applications.yml"]
[2024-08-18][10:07:13][seelen_ui::seelen_weg::icon_extractor][TRACE] Extracting icon for "ONENOTEM.EXE"
[2024-08-18][10:07:13][seelen_ui::seelen_weg::icon_extractor][TRACE] Extracting icon for "ONENOTEM.EXE"
```

**Comment by eythaann** [comment]
@Tsubajashi it was solved on 2e156836a44ad03189cb70279dfd88f5e9d04548
it is been deploying now

**Comment by eythaann** [comment]
I will close this issue the crash on open was fixed and other issue related with crash on settings change too.
Any other issue create a new ticket or send a message to the discord channel: https://discord.gg/ABfASx5ZAJ.
